libraries/AP_AS5600_AOA/AS5600_AOA.cpp

```diff
@@ -0,0 +1,177 @@
+/****************************************************
+ * AS5600 Angle of Attack Class for Ardupilot platform
+ * Author: Cole Mero
+ * Date: 15 Feb 2021
+ * File: AS5600_AOA.cpp
+ * Based on: www.ams.com
+ *
+ * Description:  This class has been designed to
+ * access the AS5600 magnetic encoder sensor to
+ * read the angle of attack and record it for experimental purposes
+ *
+ * It was adapted from the Arduino library available for the AS5600 sensor.
+ *
+***************************************************/
+
+#include <unistd.h>
+#include "AS5600_AOA.h"
+#include <AP_Common/AP_Common.h>
+
+extern const AP_HAL::HAL &hal;
+
+/**************************************************
+ * Method: AS5600_AOA
+ * In: none
+ * Out: none
+ * Description: constructor for AS5600_AOA
+***************************************************/
+ AS5600_AOA::AS5600_AOA()
+{
+    bus = 1; //Sets the bus number for the device, unclear what this number should be, trial and error to make it work
+    address = 0x36; //This is the I2C address for the device, it is set by the manufacturer
+}
+
+
+bool AS5600_AOA::init()
+{
+    dev = hal.i2c_mgr->get_device(bus, address);
+
+    if (!dev){
+        return false;
+    }
+
+    WITH_SEMAPHORE(dev->get_semaphore());
+    dev->set_speed(AP_HAL::Device::SPEED_LOW);

```

Comment:
We should be looking for a `WHOAMI` register here to make sure we're talking to who we're supposed to be talking to.